parley/src/layout/editor.rs

```diff
@@ -90,6 +90,15 @@ impl<'source> IntoIterator for SplitString<'source> {
     }
 }
 
+/// IME compose state.
+#[derive(Clone)]
+struct Compose {
+    /// Byte offsets of IME composing preedit text in the text buffer.
+    preedit_range: Range<usize>,
+    /// Whether the cursor should be shown. The IME can request to hide the cursor.
+    show_cursor: bool,
+}
```

Comment:
As a style note, I could be convinced to move both of these to `PlainEditor`... It's a trade-off between type-level correctness and ergonomics. In fact, "not-composing" could be encoded with an empty `preedit_range`, which with the current layout would shave 8 bytes off `PlainEditor`.

Reply:
`self.compose.as_ref().map(|c| c.show_cursor).unwrap_or(true)` is so ugly (even `self.compose.as_ref().is_none_or(|c| c.show_cursor)` is still pretty bad) that I think flattening this is a win. It's effectively only set in two functions anyway, so it's unlikely to get out-of-sync.

Reply:
Agreed.